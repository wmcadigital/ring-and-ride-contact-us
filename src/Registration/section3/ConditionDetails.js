import { useFormState } from "react-final-form";

import { required } from "../../common/validation";
import FormSection from "../../common/FormSection";
import TextAreaContainer from "../../common/TextAreaContainer";
import ProgressIndicator from "../../common/ProgressIndicator";
import Question from "../../common/Question";
import TextArea from "../../common/TextArea";
import getRequirementsSectionName from "./getRequirementsSectionName";

const ConditionDetails = () => {
  const stateApi = useFormState();
  const formValues = stateApi.values;

  const error = stateApi.submitFailed ? stateApi.errors?.conditionDetail : null;

  const detailsLabel =
    formValues["registerForYourself"] === "yes"
      ? "Details of your condition(s)"
      : `Details of ${formValues["firstName"]}'s condition(s)`;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 3 of 3"
        sectionName={getRequirementsSectionName(stateApi)}
      />
      <TextAreaContainer>
        <Question text="Please describe the conditions(s)" />
        <TextArea
          fieldName="conditionDetail"
          label={detailsLabel}
          rows={8}
          validation={required}
          error={error}
        />
      </TextAreaContainer>
    </FormSection>
  );
};

export default ConditionDetails;
