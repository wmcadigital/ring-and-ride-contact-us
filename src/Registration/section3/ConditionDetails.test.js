import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import ConditionDetails from "./ConditionDetails";

jest.mock("react-final-form");

describe("ConditionDetails - snapshots", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<ConditionDetails />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "no",
        firstName: "Test",
        lastName: "Testerson",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<ConditionDetails />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
      },
      submitFailed: true,
      errors: { conditionDetail: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<ConditionDetails />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
