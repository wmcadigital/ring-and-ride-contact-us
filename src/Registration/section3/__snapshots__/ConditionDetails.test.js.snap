// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionDetails - snapshots individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Applicant requirements"
    sectionPosition="Section 3 of 3"
  />
  <TextAreaContainer>
    <Question
      text="Please describe the conditions(s)"
    />
    <TextArea
      error={null}
      fieldName="conditionDetail"
      label="Details of Test's condition(s)"
      rows={8}
      validation={[Function]}
    />
  </TextAreaContainer>
</FormSection>
`;

exports[`ConditionDetails - snapshots individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Your requirements"
    sectionPosition="Section 3 of 3"
  />
  <TextAreaContainer>
    <Question
      text="Please describe the conditions(s)"
    />
    <TextArea
      error={null}
      fieldName="conditionDetail"
      label="Details of your condition(s)"
      rows={8}
      validation={[Function]}
    />
  </TextAreaContainer>
</FormSection>
`;

exports[`ConditionDetails - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Your requirements"
    sectionPosition="Section 3 of 3"
  />
  <TextAreaContainer>
    <Question
      text="Please describe the conditions(s)"
    />
    <TextArea
      error="Required"
      fieldName="conditionDetail"
      label="Details of your condition(s)"
      rows={8}
      validation={[Function]}
    />
  </TextAreaContainer>
</FormSection>
`;
